processing/src/main/java/org/apache/druid/query/ResourceLimitExceededException.java

```diff
@@ -35,9 +34,14 @@
 {
   public static final String ERROR_CODE = "Resource limit exceeded";
 
-  public ResourceLimitExceededException(String message, Object... arguments)
+  public ResourceLimitExceededException(String message)
```

Comment:
hmm, the built in format seemed useful and cleaner than having callers deal with it everywhere, and consistent with many of our other exceptions, why this change?

Reply:
I made a mistake in my previous PR that creates a `ResourceLimitExceededException` by calling `new ResourceLimitExceededException(errorCode, message, errorClass, host)` which actually calls `ResourceLimitExceededException(String message, Object... arguments)`. I didn't notice my mistake until I add a test for it because Intellij says you are cool. This change was to avoid such mistakes, but I agree it is useful. I added a static constructor that does the same.